### PR TITLE
Shiny new serialization system, now with automagic vertex deletion

### PR DESCRIPTION
**Pull request by helgridly**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 1]
```diff
@@ -38,6 +37,8 @@ object VertexSchema {
   val allClasses = Seq(Workspace, Entity, MethodConfig, MethodRepoConfig, MethodRepoMethod, Submission, Workflow, WorkflowFailure, Map)
 
   def vertexClassOf[T: TypeTag]: String = typeOf[T].typeSymbol.name.decodedName.toString
+  def vertexClassOf(tpe: Type): String  = tpe.typeSymbol.name.decodedName.toString
+  //def vertexClassToType(str: String): Type = ru.runtimeMirror(getClass.getClassLoader).staticClass(str.replace("^", ".")).selfType
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: This is unused but I want to leave it here for a mysterious future where it might be needed.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala [hunk 2]
```diff
@@ -45,14 +45,18 @@ case class Workflow(
   statusLastChangedDate: DateTime,
   workflowEntity: AttributeEntityReference,
   messages: Seq[AttributeString] = Seq.empty
-)
+) extends DomainObject {
+  def idField = "workflowId"
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: I think you can make these val

Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: or maybe not with the asMethod thing

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 3]
```diff
@@ -49,12 +50,34 @@ object VertexSchema {
...
+  val allEdgeRelations = Seq(Own, Ref)
+
+  def getEdgeRelation(label: String): EdgeRelationType = {
+    toEdgeRelation(label.split(sep).head)
+  }
+
+  def toEdgeRelation(prefix: String): EdgeRelationType = {
+    allEdgeRelations.filter( er => prefix == er.prefix ).head
+  }
+  def stripEdgeRelation(str: String): String = str.split(sep, 2).last
```
Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: isnt this stripping the label off the edge relation?  not sure naming is good if so

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: It splits the relation type - OWN or REF - from the edge label.

Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: wouldnt that be head?

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: sigh. no. It _removes_ the relation type from the edge label. An edge label `foo` gets rewritten to `OWN_foo`. This function returns the "original name" of the label, i.e. without the `OWN_`.

Comment (abaumann) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: ok i understand the meaning of why its called that now.  dont sigh at me

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: you can't tell me what to do you're not even my real dad

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 4]
```diff
@@ -49,12 +50,34 @@ object VertexSchema {
   }
 }
 
+object EdgeSchema {
+  val sep = "_"
+  sealed trait EdgeRelationType {
+    def prefix: String
+    def toLabel(label:String) = prefix + sep + label
```
Comment (mattsooknah) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Minor point, but it's confusing to have a ```toLabel``` function that takes ```label```. Especially since ```label``` is also used below to denote the whole thing. Maybe ```suffix``` instead?

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Legit.

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 5]
```diff
@@ -49,12 +50,34 @@ object VertexSchema {
...
+  }
+  val allEdgeRelations = Seq(Own, Ref)
+
+  def getEdgeRelation(label: String): EdgeRelationType = {
+    toEdgeRelation(label.split(sep).head)
+  }
+
+  def toEdgeRelation(prefix: String): EdgeRelationType = {
+    allEdgeRelations.filter( er => prefix == er.prefix ).head
+  }
```
Comment (mattsooknah) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Instead of all this direct string manipulation, why not have an ```EdgeDescriptor``` case class that contains an ```EdgeRelationType``` and a ```String``` for the suffix, and has a ```toEdgeLabel``` method that joins them?

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: `EdgeDescriptor(EdgeSchema.Ref, "foo").toEdgeLabel` is a bit wordy, no?

Comment (mattsooknah) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Yeah, I suppose it is. I guess what would be nice is the ability to serialize classes into edge labels, so that any arbitrary information can be encoded there ... but that way leads to madness. So, direct string manipulation is fine.

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: I briefly considered using annotations to distinguish ref/own types, but figured that'd mean instant pariah.
